src/vs/workbench/contrib/terminal/browser/terminalActions.ts

```diff
@@ -154,15 +154,47 @@ export function registerTerminalActions() {
 				},
 			});
 		}
-		async run(accessor: ServicesAccessor, profile?: ITerminalProfile) {
+		async run(accessor: ServicesAccessor, event: unknown, profile?: ITerminalProfile) {
 			const terminalService = accessor.get(ITerminalService);
-			if (profile) {
-				const instance = terminalService.createTerminal(profile);
-				terminalService.setActiveInstance(instance);
-				return terminalService.showPanel(true);
-			} else {
-				await terminalService.showProfileQuickPick('createInstance');
+			const workspaceContextService = accessor.get(IWorkspaceContextService);
+			const commandService = accessor.get(ICommandService);
+			const folders = workspaceContextService.getWorkspace().folders;
+			if (event instanceof MouseEvent && (event.altKey || event.ctrlKey)) {
+				const activeInstance = terminalService.getActiveInstance();
```

Comment:
when it's a split do we want to assume the cwd should be used?

Reply:
👍